cypress/e2e/GUI/gui.cy.js

```diff
@@ -0,0 +1,165 @@
+            .find('input[type="email"]')
+            .should('have.attr', 'required')
+        cy.getByClassThatStartsWith('Messenger_form')
+            .find('textarea')
+            .should('have.attr', 'required')
+    })
+
+    it('Mostra e oculta uma mensagem de sucesso ao enviar o formulario do messenger com sucesso', () => {
+        cy.getByClassThatStartsWith('Messenger_openCloseButton').click()
+        cy.clock()
+        cy.get('input[id="messenger-name"]').type('Joe')
+        cy.get('input[id="email"]').type('test@example.com')
+        cy.get('textarea[id="message"]').type('Test')
+
+        cy.get('button[type="submit"]').click()
+
+        cy.contains('Your message has been sent').should('be.visible')
+        cy.tick(3000)
+        cy.contains('div.Messenger_success_nqnL1').should('not.exist')
+    })
+
+    it('Limpa todos os campos do formulario do messenger ao preenche-los, fechar o messenger e abri-lo novamente', () => {
+        cy.getByClassThatStartsWith('Messenger_openCloseButton').click()
+        cy.get('input[id="messenger-name"]').type('Joe')
+        cy.get('input[id="email"]').type('test@example.com')
+        cy.get('textarea[id="message"]').type('Test')
+
+        cy.getByClassThatStartsWith('Messenger_openCloseButton').click()
+        cy.getByClassThatStartsWith('Messenger_openCloseButton').click()
+
+        cy.get('input[id="messenger-name"]').should('have.value', '')
+        cy.get('input[id="email"]').should('have.value', '')
+        cy.get('textarea[id="message"]').should('have.value', '')
+    })
+
+    context('Testando com dados Mockados', () => {
+        beforeEach(() => {
+            cy.intercept(
+                {
+                    method: 'GET',
+                    pathname: '**/customers',
+                    query: {
+                        page: '1',
+                        limit: '10',
+                        size: 'All',
+                        industry: 'All'
+                    }
+                },
+                { fixtures: 'companies' }
+            ).as('getCompanies')
+            cy.visit('/')
+            cy.wait('@getCompanies')
+        })
+
+        it('Verifica se e possivel acessar os detalhes do cliente', () => {
+            cy.contains('button', 'View').click()
+            cy.contains('Customer Details').should('be.visible')
+            cy.contains('Test 1').should('be.visible')
+        })
+
+        it('Verifica se e possivel expandir e recolher o endereco do cliente', () => {
+            cy.contains('button', 'View').click()
+            cy.contains('Test 1').should('be.visible')
+            cy.contains('Show address').click()
+            cy.contains('Test Street 1').should('be.visible')
+            cy.contains('Hide address').should('be.visible')
+            cy.contains('Hide address').click()
+            cy.contains('Show address').should('be.visible')
+            cy.contains('Test Street 1').should('not.be.visible')
+        })
+    })
+})
```

Comment:
Aqui tbm deverias adicionar uma nova linha ao final do arquivo.

Reply:
Creio que aqui já há uma linha em branco no final visto que o ícone de `No newline at end of file` não é exibido.